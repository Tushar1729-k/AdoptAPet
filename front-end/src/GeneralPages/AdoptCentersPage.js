import React from 'react'
import { Row, Table, Col } from 'react-bootstrap'
import {  useState, useEffect } from 'react'
import adoptionCenters from '../Data/AdoptionCenters.json'
import Paginate from '../Components/Pagination'
import axios from 'axios'
import { Link } from "react-router-dom"
import PropTypes from 'prop-types'

const AdoptCentersPage = ({fetchPage}) => {
    const [allCenters, setAllCenters] = useState([])
    const [centersPerPage, setCentersPerPage] = useState(10)
    const [currentPage, setCurrentPage] = useState(1)
    const [isLoading, setIsLoading] = useState(false)

    const fetchCenters = async (pageNum) => {
        setIsLoading(true)
        const res = await axios.get(`https://api.adoptapet.me/ac?page=${pageNum}`)
        setAllCenters(res.data.page)
        setIsLoading(false)
        setCentersPerPage(res.data.count)
    }

    useEffect(() => {
        fetchCenters(1)
    }, [])
    const paginate = (num) => {
        setCurrentPage(num)
        fetchCenters(num)
    }
    const whichCenterPage = (type, num) => {
        fetchPage(type, num)
    }
    return (
        <div style={{paddingLeft: '10vw', paddingRight: '10vw'}}>
            <Row xs={1}>
                <Col>
                    <h2>Adoption Centers</h2>
                    <h6>20 adoption centers, page {currentPage}/5</h6>
                    {isLoading && <h4>Loading...</h4>}
                </Col>
                <Col>
                    <Paginate totalItems={100} itemsPerPage={20} paginate={paginate} />
                </Col>
            </Row>
            <div style={{ paddingTop: '2vh'}}>
            <Table striped bordered hover size="sm" responsive>
                <thead>
                    <tr>
                    <th>Name</th>
                    <th>City</th>
                    <th>State</th>
                    <th>Zip</th>
                    <th>Services</th>
                    </tr>
                </thead>
                <tbody>
                {allCenters.map((center, index) => (
                    <tr key={index}>
                    <Link to={`/acmodel/${center.api_id}`} style={{ textDecoration: 'none'}} onClick={() => whichCenterPage("ac", center.api_id)}>
                        <td>{center.name}</td>
                    </Link>
                    <td>{center.city}</td>
                    <td>{center.state}</td>
                    <td>{center.zipcode}</td>
                    <td>{center.services}</td>
                    </tr>
                ))}
                </tbody>
                </Table>
                </div>
        </div>
    )
}

AdoptCentersPage.propTypes = {
    fetchPage: PropTypes.func
}

export default AdoptCentersPage
