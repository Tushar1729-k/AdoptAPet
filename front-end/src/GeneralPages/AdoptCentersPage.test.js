import React from 'react';
import renderer from 'react-test-renderer';
import AdoptCentersPage from './AdoptCentersPage';
import axios from 'axios'
import AdoptCentersPageTestData from '../Data/AdoptCentersPageTestData.json'

test("Pets Model Page Properly Rendered", () => {
    jest.mock('axios')
    axios.get = jest.fn()
    const resp = AdoptCentersPageTestData;
    axios.get.mockResolvedValue(resp);
    const component = renderer.create(
        <AdoptCentersPage/>,
      );
      let tree = component.toJSON();
      expect(tree).toMatchSnapshot();
})