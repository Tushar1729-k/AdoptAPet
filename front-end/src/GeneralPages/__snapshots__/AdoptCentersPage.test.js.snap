// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pets Model Page Properly Rendered 1`] = `
<div
  style={
    Object {
      "paddingLeft": "10vw",
      "paddingRight": "10vw",
    }
  }
>
  <div
    className="row row-cols-1"
  >
    <div
      className="col"
    >
      <h2>
        Adoption Centers
      </h2>
      <h6>
        20 adoption centers, page 
        1
        /5
      </h6>
    </div>
    <div
      className="col"
    >
      <div>
        <ul
          className="pagination"
        >
          <li
            className="page-item"
          >
            <a
              className="page-link"
              onClick={[Function]}
              onKeyDown={[Function]}
              role="button"
              tabIndex={0}
            >
              <span
                aria-hidden="true"
              >
                ‹
              </span>
              <span
                className="visually-hidden"
              >
                Previous
              </span>
            </a>
          </li>
          <li
            className="page-item active"
          >
            <span
              className="page-link"
              disabled={false}
              onClick={[Function]}
            >
              1
              <span
                className="visually-hidden"
              >
                (current)
              </span>
            </span>
          </li>
          <li
            className="page-item"
          >
            <a
              className="page-link"
              onClick={[Function]}
              onKeyDown={[Function]}
              role="button"
              tabIndex={0}
            >
              2
            </a>
          </li>
          <li
            className="page-item"
          >
            <a
              className="page-link"
              onClick={[Function]}
              onKeyDown={[Function]}
              role="button"
              tabIndex={0}
            >
              3
            </a>
          </li>
          <li
            className="page-item"
          >
            <a
              className="page-link"
              onClick={[Function]}
              onKeyDown={[Function]}
              role="button"
              tabIndex={0}
            >
              4
            </a>
          </li>
          <li
            className="page-item"
          >
            <a
              className="page-link"
              onClick={[Function]}
              onKeyDown={[Function]}
              role="button"
              tabIndex={0}
            >
              5
            </a>
          </li>
          <li
            className="page-item"
          >
            <a
              className="page-link"
              onClick={[Function]}
              onKeyDown={[Function]}
              role="button"
              tabIndex={0}
            >
              <span
                aria-hidden="true"
              >
                …
              </span>
              <span
                className="visually-hidden"
              >
                More
              </span>
            </a>
          </li>
          <li
            className="page-item"
          >
            <a
              className="page-link"
              onClick={[Function]}
              onKeyDown={[Function]}
              role="button"
              tabIndex={0}
            >
              <span
                aria-hidden="true"
              >
                ›
              </span>
              <span
                className="visually-hidden"
              >
                Next
              </span>
            </a>
          </li>
        </ul>
      </div>
    </div>
  </div>
  <div
    style={
      Object {
        "paddingTop": "2vh",
      }
    }
  >
    <div
      className="table-responsive"
    >
      <table
        className="table table-sm table-striped table-bordered table-hover"
      >
        <thead>
          <tr>
            <th>
              Name
            </th>
            <th>
              City
            </th>
            <th>
              State
            </th>
            <th>
              Zip
            </th>
            <th>
              Services
            </th>
          </tr>
        </thead>
        <tbody />
      </table>
    </div>
  </div>
</div>
`;
